extensions/interactions/MathExpressionInput/MathExpressionInput.js

```diff
@@ -30,13 +30,127 @@ oppia.directive('oppiaInteractiveMathExpressionInput', [
       templateUrl: 'interaction/MathExpressionInput',
       controller: [
         '$scope', '$attrs', '$timeout', '$element', 'LABEL_FOR_CLEARING_FOCUS',
-        function($scope, $attrs, $timeout, $element, LABEL_FOR_CLEARING_FOCUS) {
+        'oppiaDebouncer', 'deviceInfoService',
+        function($scope, $attrs, $timeout, $element, LABEL_FOR_CLEARING_FOCUS,
+          oppiaDebouncer, deviceInfoService) {
           var guppyDivElt = $element[0].querySelector('.guppy-div');
+
+          /**
+           * Adds a button overlay and invisible text field used to bring up
+           * the keyboard on mobile devices.
+           *
+           * TODO: On iOS/Safari, keyboard may only appear on first press.
```

Comment:
Format should be `TODO(Oishikatta):`, so that people know who to ask for clarification.